src/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/Modules/GetTypeCoreCache.cs

```diff
@@ -0,0 +1,225 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.Reflection.TypeLoading
+{
+    internal sealed class GetTypeCoreCache
```

Comment:
Curious why this modified clone of Hashtable?

Reply:
Has I noticed that too and assumed it was an optimization. I'll come back to this later.

Reply:
It appears it is intended to work on two spans at the same time, and without having to wrap them:
```
ReadOnlySpan<byte> ns, ReadOnlySpan<byte> name
...
IsTypeNameEqual(ReadOnlySpan<byte> ns, ReadOnlySpan<byte> name)
```